openedx/core/djangoapps/auth_exchange/tests/utils.py

```diff
@@ -86,16 +86,19 @@ def test_invalid_acess_token(self):
 
     def test_no_linked_user(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
         self._setup_provider_response(success=True)
         self._assert_error(self.data, "invalid_grant", "access_token is not valid")
 
     def test_user_automatically_linked_by_email(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
         self._setup_provider_response(success=True, email=self.user.email)
         self._assert_success(self.data, expected_scopes=[])
 
     def test_inactive_user_not_automatically_linked(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
```

Comment:
@e-kolpakov So PSA isn't deleting the Partial objects on its own; we have a lot of freedom with them. From what I remember while doing this, these particular tests weren't playing nice until I added pre-deletions. If you're still curious, I can do a 2nd round of investigation on this and see if it's still necessary or what precisely goes wrong without them.